blaze/expr/tests/test_reductions.py

```diff
@@ -97,3 +97,33 @@ def test_max_min_on_datetime_and_timedelta(reduc, measure):
 
 def test_reduction_naming_with_generated_leaves():
     assert symbol('_', 'var * float64').sum()._name == 'sum'
+
+
+@pytest.mark.parametrize(
+    ['reduction', 'group_by', 'sort_by', 'preceding', 'following'],
+    list(product(
+        ['sum', 'mean', 'count'],
+        list('ab'),
+        list('ab'),
+        [None, 0, 3],
+        [None, 0, 3]
+    ))
+)
+def test_basic_over(reduction, group_by, sort_by, preceding, following):
+    t = symbol('t', 'var * {a: int64, b: ?float64}')
+    expr = getattr(t.b, reduction)().over(
+        group_by=group_by,
+        sort_by=sort_by,
```

Comment:
I could see suporting both options. This would allow you to pass all that information in at the construction of the `over` call _or_ chain new behavior as you go. I Like the way the current `over` call lets you express all the relevant inforation quickly. Right now the only fear I would have with the method chaining would be that the `sort` top level function creates `Sort` objects, not `WindowSort` objects. Currently `a.b()` is always the same as `b(a)` so I am not thrilled about changing that.